libcamera-sys/c_api/camera.cpp

```diff
@@ -67,7 +69,11 @@ const libcamera_control_list_t *libcamera_camera_properties(const libcamera_came
 }
 
 libcamera_camera_configuration_t *libcamera_camera_generate_configuration(libcamera_camera_t *cam, const enum libcamera_stream_role *roles, size_t role_count) {
+#if LIBCAMERA_VERSION_MAJOR == 0 && LIBCAMERA_VERSION_MINOR == 0
     libcamera::StreamRoles roles_vec((libcamera::StreamRole*)roles, (libcamera::StreamRole*)roles + role_count);
```

Comment:
in fact, now I think about it you can entirely get rid of this conditional too by removing the use of libcamera::StreamRoles here and using 
```
    std::vector<StreamRole> roles_vec((libcamera::StreamRole*)roles, (libcamera::StreamRole*)roles + role_count);
    return cam->get()->generateConfiguration(roles_vec).release();
```


```

Reply:
But I'm sure you could probably still use a utils::Span to do the same here for both cases as well! (utils::Span is likely more efficient than a std::vector<>)

Reply:
ok, that's fine for me. I changed it back to std::vector and also went back to c++17 (c++20 was only needed for span).